libs/core/kiln_ai/adapters/extractors/gemini_extractor.py

```diff
@@ -0,0 +1,152 @@
+from enum import Enum
+
+from google import genai
+from google.genai import types
+from pydantic import Field
+
+from kiln_ai.adapters.extractors.base_extractor import (
+    BaseExtractor,
+    BaseExtractorConfig,
+    ExtractionFormat,
+    ExtractionOutput,
+    FileInfoInternal,
+)
+
+
+class Kind(Enum):
+    DOCUMENT = "document"
+
+    # NOTE: maybe remove the these ones for now as we don't use them and the mimetypes list for them
+    # might evolve by the time we do add support for them
+    IMAGE = "image"
+    VIDEO = "video"
+    AUDIO = "audio"
+
+
+# docs list out supported formats:
+# - https://ai.google.dev/gemini-api/docs/document-processing#supported-formats
+# - https://ai.google.dev/gemini-api/docs/image-understanding#supported-formats
+# - https://ai.google.dev/gemini-api/docs/video-understanding#supported-formats
+# - https://ai.google.dev/gemini-api/docs/audio#supported-formats
+MIME_TYPES_SUPPORTED = {
+    Kind.DOCUMENT: [
+        "application/pdf",
+        "application/x-javascript",
+        "text/javascript",
+        "application/x-python",
+        "text/x-python",
+        "text/plain",
+        "text/markdown",  # not officially listed, but works
+        "text/html",
+        "text/css",
+        "text/md",
+        "text/csv",
+        "text/xml",
+        "text/rtf",
+    ],
+    Kind.IMAGE: [
+        "image/png",
+        "image/jpeg",
+        "image/webp",
+        "image/heic",
+        "image/heif",
+    ],
+    Kind.VIDEO: [
+        "video/mp4",
+        "video/mpeg",
+        "video/mov",
+        "video/avi",
+        "video/x-flv",
+        "video/mpg",
+        "video/webm",
+        "video/wmv",
+        "video/3gpp",
+    ],
+    Kind.AUDIO: [
+        "audio/wav",
+        "audio/mp3",
+        "audio/mpeg",  # not explicitly supported, but mimetypes stdlib returns audio/mpeg for MP3
+        "audio/aiff",
+        "audio/aac",
+        "audio/ogg",
+        "audio/flac",
+    ],
+}
+
+
+class GeminiExtractorConfig(BaseExtractorConfig):
+    default_prompt: str = Field(
+        description="The default prompt to use for the extractor."
+    )
+    custom_prompt: str | None = Field(
+        default=None,
+        description="A custom prompt provided by the user to use for the extractor. It takes precedence over the prompt_for_kind and default_prompt.",
+    )
+    prompt_for_kind: dict[Kind, str] = Field(
```

Comment:
Let's collapse all 3, just keeping "prompt_for_kind"? Will simplify downstream usage. 

We don't need/want default/custom logic at the config level. The fact that there's a default will be handled in config init, or even just the UI. But the caller always just uses the saved prompt.

We should add a validator which checks all values are set in prompt_for_kind (doc, vid, etc), but the initializer can use defaults for any not set manually, which will keep usage easy.

Reply:
The prompt always comes from the config, so it's always "saved" (since configs are usually persisted prompt_for_kind is "saved", is what I was trying to say). The adapter just uses prompt_for_kind. If there's default/custom logic, that's when then ExtratorConfig is created - but doesn't need to be saved beyond that and the adapter doesn't need to be aware of it.